packages/synapse-react-client/src/components/SqlDefinedTableEditor/SqlDefinedTableEditorModal.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react'
+import ConfirmationDialog from '../ConfirmationDialog'
+import SqlDefinedTableEditor from './SqlDefinedTableEditor'
+import { useGetEntity, useUpdateEntity } from '../../synapse-queries'
+import { convertToEntityType } from '../../utils/functions/EntityTypeUtils'
+import { MaterializedView, VirtualTable } from '@sage-bionetworks/synapse-types'
+import { Alert } from '@mui/material'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+
+export type SqlDefinedTableEditorModalProps = {
+  open: boolean
+  entityId: string
+  onUpdate: () => void
+  onCancel: () => void
+}
+export default function SqlDefinedTableEditorModal(
+  props: SqlDefinedTableEditorModalProps,
+) {
+  const { open, entityId, onCancel, onUpdate } = props
+  const {
+    data: entity,
+    isLoading: isEntityLoading,
+    error: entityError,
+  } = useGetEntity<MaterializedView | VirtualTable>(entityId)
+  const [sql, setSql] = useState('')
+
+  useEffect(() => {
+    if (entity) {
+      setSql(entity.definingSQL)
+    }
+  }, [entity])
+
+  const type = entity ? convertToEntityType(entity.concreteType) : undefined
+  const {
+    mutate,
+    isLoading: isUpdateLoading,
+    error: updateError,
+  } = useUpdateEntity<MaterializedView | VirtualTable>({ onSuccess: onUpdate })
+
```

Comment:
Let's consolidate `isLoading` and `error`s up here instead of inline
```suggestion
  const isLoading = isEntityLoading || isUpdateLoading
  const error = entityError || updateError
```